icepyx/core/query.py

```diff
@@ -741,9 +741,7 @@ def granules(self):
         <icepyx.core.granules.Granules at [location]>
         """
 
-        if not hasattr(self, "_granules"):
-            self._granules = Granules()
-        elif self._granules is None:
+        if not hasattr(self, "_granules") or self._granules is None:
```

Comment:
Huge value from this type of simplification; it would be easy to introduce a bug by changing only one branch of the existing conditional and forgetting the other.